docker-entrypoint.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+SHARED_DIR="/var/lib/goofys"
+
+trap "umount ${SHARED_DIR}" SIGKILL SIGTERM SIGHUP SIGINT EXIT
+
+mount --make-shared $SHARED_DIR
+OPTS=" -o allow_other -f"
+
+if ! test -z "$GOOFYS_GID"; then
+  groupadd -g $GOOFYS_GID goofys
+  OPTS="${OPTS} --gid ${GOOFYS_GID}"
+fi
+
+if ! test -z "$GOOFYS_UID"; then
+  useradd -u $GOOFYS_UID goofys
+  OPTS="${OPTS} --uid ${GOOFYS_UID}"
+fi
+
+if ! test -z "$GOOFYS_ENDPOINT"; then
+  OPTS="${OPTS} --endpoint ${GOOFYS_ENDPOINT}"
+fi
+
+$GOPATH/bin/goofys $OPTS $EXTRA_OPTS $BUCKETNAME $SHARED_DIR &
+
+wait
```

Comment:
It's for the `trap` to work properly, to make sure the `umount` happens properly before the process is killed. In my experience (at least with s3fs), if the process is killed to quickly without making sure the mount point is unmounted, then it's not possible to restart the container (because the mount point is still used).